keyserver/listener.go

```diff
@@ -0,0 +1,127 @@
+package keyserver
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net"
+	"time"
+
+	. "github.com/coniks-sys/coniks-go/protocol"
+)
+
+func (server *ConiksServer) listenForRequests(ln net.Listener, handler func(msg []byte) ([]byte, error)) {
```

Comment:
You can then add a second wait group just for the accepted requests and call Wait on that group after <-server.stop so that they all get handled cleanly, then return to close the connection.